indico/modules/events/contributions/operations.py

```diff
@@ -76,6 +81,11 @@ def update_contribution(contrib, data):
     """
     rv = {'unscheduled': False, 'undo_unschedule': None}
     current_session_block = contrib.session_block
+    start_dt = data.pop('start_date', None)
+    if start_dt is not None:
+        if not contrib.timetable_entry:
```

Comment:
Why don't you use `create_timetable_entry`/`update_timetable_entry`? The point of those functions is that there is a single place so we don't have schedule modifications all over (it'd be a HUGE pain in the ass like it is in the current legacy code since it's so easy to miss a place where it's modified when changing things)